### PR TITLE
Use AssetManagerAndQuarantinedFileStorage for attachments

### PR DESCRIPTION
**Pull request by chrisroos**
This PR configures the `AttachmentUploader` to store attachments on the file system and in Asset Manager. This addresses https://github.com/alphagov/asset-manager/issues/440.

I've tested this change as working in the development vm. I created a consultation and uploaded a single attachment and a single image. The attachment was created in the draft state while the images were created in a non-draft state:

We're not going to be ready to serve these attachment assets from Asset Manager for a while so marking them as draft seems like the safest option in the meantime.

### CODE REVIEW

File: app/workers/asset_manager_create_whitehall_asset_worker.rb [hunk 1]
```diff
@@ -1,7 +1,9 @@
 class AssetManagerCreateWhitehallAssetWorker < WorkerBase
   def perform(file_path, legacy_url_path)
     file = File.open(file_path)
-    Services.asset_manager.create_whitehall_asset(file: file, legacy_url_path: legacy_url_path)
+    asset_options = {file: file, legacy_url_path: legacy_url_path}
```
Comment (floehopper) [inline app/workers/asset_manager_create_whitehall_asset_worker.rb]: Very minor: It's usual to have spaces after the opening brace and before the closing brace of a Hash. I would've expected Rubocop to complain about this.

File: app/workers/asset_manager_create_whitehall_asset_worker.rb [hunk 2]
```diff
@@ -1,7 +1,9 @@
...
+    asset_options[:draft] = true if legacy_url_path =~ /attachment_data/
```
Comment (floehopper) [inline app/workers/asset_manager_create_whitehall_asset_worker.rb]: Minor: I'm not keen on trailing `if` statements - I think the code is clearer if you use the block form.

File: app/workers/asset_manager_create_whitehall_asset_worker.rb [hunk 3]
```diff
@@ -1,8 +1,8 @@
 class AssetManagerCreateWhitehallAssetWorker < WorkerBase
   def perform(file_path, legacy_url_path)
     file = File.open(file_path)
-    asset_options = {file: file, legacy_url_path: legacy_url_path}
-    asset_options[:draft] = true if legacy_url_path =~ /attachment_data/
+    asset_options = { file: file, legacy_url_path: legacy_url_path }
+    asset_options[:draft] = true if legacy_url_path.match?(/attachment_data/)
```
Comment (floehopper) [inline app/workers/asset_manager_create_whitehall_asset_worker.rb]: Minor: These changes seem orthogonal to the commit note. I think they'd be better combined into the first commit.

File: test/unit/whitehall_uploader_test.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class WhitehallUploaderTest < ActiveSupport::TestCase
+  test 'indicates that assets are protected' do
```
Comment (floehopper) [inline test/unit/whitehall_uploader_test.rb]: Typo: I think this should be "indicates that assets are *not* protected"

**Comment by floehopper** [comment]
@chrisroos: One other thought. It would be worth explaining somewhere what the rationale is for marking *all* new attachments as draft, irrespective of their state in Whitehall. Presumably we'll need to update the draft status of these assets when we've implemented https://github.com/alphagov/asset-manager/issues/447.

File: app/workers/asset_manager_create_whitehall_asset_worker.rb (hunk 1 above)
Comment (chrisroos) [reply app/workers/asset_manager_create_whitehall_asset_worker.rb]: Rubocop did complain! I fixed it but put it in the subsequent commit by mistake. I've now updated this commit.

File: app/workers/asset_manager_create_whitehall_asset_worker.rb (hunk 2 above)
Comment (chrisroos) [reply app/workers/asset_manager_create_whitehall_asset_worker.rb]: I generally agree if it's a long line but I'm happy that this is short enough not to be too unreadable so I'm going to leave it as is.

File: app/workers/asset_manager_create_whitehall_asset_worker.rb (hunk 3 above)
Comment (chrisroos) [reply app/workers/asset_manager_create_whitehall_asset_worker.rb]: Good spot. I incorrectly added it to this commit. I've now moved it to the previous commit.

File: test/unit/whitehall_uploader_test.rb (hunk 4 above)
Comment (chrisroos) [reply test/unit/whitehall_uploader_test.rb]: Correct! I've updated the test name.

**Comment by chrisroos** [comment]
> One other minor thought: it would be nice to use keyword arguments for worker perform method now there are three of them. However, I imagine the switchover might cause problems with queued jobs, so perhaps it's not worth it.

I agree that keyword args would be better but that switching over would be a bit painful so I'm not going to do anything about this for now.

**Comment by chrisroos** [comment]
> One other thought. It would be worth explaining somewhere what the rationale is for marking all new attachments as draft, irrespective of their state in Whitehall.

I've added a note to the commit that marks these attachments as draft in asset-manager. I've added the same note to the description of this PR.

> Presumably we'll need to update the draft status of these assets when we've implemented alphagov/asset-manager#447.

Yes. Or at least before we finally switch the nginx config to serve these attachments.

**Comment by chrisroos** [comment]
Thanks for the review, @floehopper. I've addressed your comments, rebased on master and force pushed in preparation for merging.
